File: C++/zasieg.cpp
/*
 
 */


#include <iostream>
using namespace std;

int suma = 0; //zasięg globalny

void drukuj (int liczba) {
    cout << suma << endl ;
    liczba += 7;
    cout << &liczba << endl;
    cout << liczba << endl;
}
void drukuj2(int &liczba) {
    cout << suma << endl ;
    liczba += 7;
    cout << &liczba<< endl;
    cout << liczba << endl;
    
}

int main(int argc, char **argv)
{
    int suma = 5; //zmienna lokalna
    int liczba = 3;
    suma += 10;
    cout << suma << endl;
    cout << liczba << endl;
    cout << &liczba << endl;
    //drukuj (liczba); // by value
    drukuj2 (liczba); //by reference
    cout << liczba << endl;
	return 0;
}

File: python/zasieg.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  zasieg.py
#  

suma = 10 # zmienna globalna

def drukuj():
    global suma
    print(suma)
    suma = 20
    
def drukuj2(liczba):
    liczba += 10
    print(liczba)
    return liczba
    
def main(args):
    suma = 5 # zmienna lokalna
    liczba = 3
    #print(suma)
    #drukuj()
    #print (suma)
    print(liczba)
    liczba = drukuj2(liczba)
    print(liczba)

    
    
    
    
    
    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))
